consensus/state.go

```diff
@@ -2181,7 +2186,20 @@ func (cs *State) signVote(
 	}
 
 	v := vote.ToProto()
-	err := cs.privValidator.SignVote(cs.state.ChainID, v)
+
+	// If the signedMessageType is for precommit,
+	// use our local precommit Timeout as the max wait time for getting a singed commit. The same goes for prevote.
+	var timeout time.Duration
+	if msgType == tmproto.PrecommitType {
+		timeout = cs.config.TimeoutPrecommit
+	} else if msgType == tmproto.PrevoteType {
+		timeout = cs.config.TimeoutPrevote
+	}
```

Comment:
We should make this a `switch` case with a default value.

Reply:
setting default as 1 second, this is default value of precommitType and PrevoteType.